e2e/js_image/js_image_layer.bzl

```diff
@@ -0,0 +1,239 @@
+"contains container helper functions for js_binary"
+
+load("@rules_pkg//:providers.bzl", "PackageFilegroupInfo", "PackageFilesInfo", "PackageSymlinkInfo")
+load("@rules_pkg//:pkg.bzl", "pkg_tar")
+load("@bazel_skylib//rules:write_file.bzl", "write_file")
+load("@aspect_bazel_lib//lib:paths.bzl", "to_manifest_path")
+load("@rules_python//python:defs.bzl", "py_binary")
+
+_LAUNCHER_TMPL = """
+export BAZEL_BINDIR=.
+source {executable_path}
+"""
+
+def _write_laucher(ctx, executable_path):
+    launcher = ctx.actions.declare_file("launcher.sh")
```

Comment:
`launcher.sh` should be name spaced, eg have the target name in the file or similar, allowing more than one in a package, otherwise the two generating actions will collide.